docs/rules/index.md

```diff
@@ -41,6 +40,19 @@ The `--fix` option on the [command line](../user-guide/command-line-interface#fi
 
 {% endfor %}
 
+{% if site.data.rules.deprecated %}
```

Comment:
If at any point in time there are no rules currently marked as "deprecated", this ensures an empty broken-looking section won't be rendered.